### PR TITLE
HTCONDOR-1903 fix issues that cause multiple DEFRAG daemons in a pool…

### PR DESCRIPTION
**Pull request by johnkn**
… to intefere with each other

  * default defrag daemon name using $(LOCALNAME) if one is not specified in config
  * add defrag daemon name to drain reason for drain commands
  * change various query expressions to check for the defrag daemon name in the reason
  * remove the defrag_state file since it collides and serves no useful purpose

<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/defrag/defrag.cpp [hunk 1]
```diff
@@ -128,14 +127,21 @@ void Defrag::init()
...
+	if ( ! param(m_defrag_name,"DEFRAG_NAME")) {
+		param(m_defrag_name, "LOCALNAME");
+	}
+	auto_free_ptr valid_name(build_valid_daemon_name(m_defrag_name.c_str()));
+	ASSERT( valid_name );
+	m_daemon_name = valid_name.ptr();
 
 	int old_polling_interval = m_polling_interval;
-	m_polling_interval = param_integer("DEFRAG_INTERVAL",60);
+	m_polling_interval = param_integer("DEFRAG_INTERVAL",600);
```
Comment (Todd-L-Miller) [inline src/defrag/defrag.cpp:144]: Did you intend to change this?  (You didn't mention it in the ticket.  It seems like a sensible change, though.)

File: src/defrag/defrag.cpp [hunk 2]
```diff
@@ -788,7 +815,7 @@ void Defrag::poll()
 	dprintf(D_ALWAYS,"Looking for %d machines to drain.\n",num_to_drain);
 
 	ClassAdList startdAds;
-	std::string requirements = "PartitionableSlot && Offline =!= true && Draining =!= true";
+	std::string requirements(BASE_SLOT_CONSTRAINT " && Draining =!= true");
```
Comment (Todd-L-Miller) [inline src/defrag/defrag.cpp:818]: Would it make sense for `Draining =!= true` to be a constant defined near `BASE_SLOT_CONSTRAINT` and/or `BASE_DRAINING_CONSTRAINT`?  Seems like you might want to make sure the machines-to-drain query and the machines-not-to-drain query are consistent (or, ideally, constructed from each other).

**Review by Todd-L-Miller — APPROVED** [review]
Two minor questions.  I'll approve but not merge, in case you want to address them.

File: src/defrag/defrag.cpp (hunk 1 above)
Comment (johnkn) [reply src/defrag/defrag.cpp:144]: fixed to match the param table.

**Pull request merged**